for-developers.md

```diff
@@ -0,0 +1,275 @@
+# Developer Guide
+
+1. [Configuration Settings](#configuration-settings)
+
+2. [Batch environment setup and fre-cli](#batch-environment-setup-and-fre-cli)
+
+3. [Post-processing guide using fre-cli and cylc](#guide)
+
+# Configuration Settings
+
+`fre-workflows` has primarily been developed and tested on PPAN. The following are guidelines for developing on PPAN:
+
+## Cylc User settings
+
+The path to the Cylc binary must be added to your `$PATH` environment variable first. This can be done by modifying your PATH in your bash_profile as follows:
+
+```
+> cat /home/First.Last/.bash_profile
+# this is essential Cylc configuration below
+export PATH="${PATH}:/home/fms/local/opt/cylc/bin"
+```
+
+We need this to run on PPAN. Without a cylc binary in your `$PATH` variable, the initial setup of the workflow server (check terminology) fails, and we cannot edit the environment with a module load or similar until it is running.
+
+> ** _NOTE:_** Adding a link to the cylc binary in a directory that is default in user's $PATH is desired, this is being persued.
+
+
+## Experiment settings
+
+The current organization of the yaml files comprises of the model, settings, and post-processing configuration files, where the model yaml holds paths that point to the settings and post-processing yamls.
+
+For more information on the yaml framework, see [fre-cli's README and documentation on the yaml files](https://noaa-gfdl.readthedocs.io/projects/fre-cli/en/latest/usage.html#model-yaml).
+
+The `settings.yaml` contains necessary experiment-specific information. You'll see reference to some of the settings in this file throughout the workflow as variables.
+
+Example `settings.yaml`:
+
+```
+> cat yaml_workflow/pp/settings.yaml:
+
+directories:
+  history_dir: "/archive/$USER/CMIP7/ESM4/DEV/ESM4.5_candidateA/ppan-prod-openmp/history"
+  pp_dir: "/archive/$USER/CMIP7/ESM4/DEV/ESM4.5_candidateA/ppan-prod-openmp/pp"
+  analysis_dir: "/nbhome/$USER/CMIP7/ESM4/DEV/ESM4.5_candidateA"
+  ptmp_dir: "/xtmp/$USER/ptmp"
+postprocess:
+  settings:
+    site: "ppan"
+    history_segment: "P1Y"
+    pp_start: 0002
+    pp_stop: 0003
+    pp_chunks: ["P5Y"]
+    pp_grid_spec: "/work/Niki.Zadeh/mosaic_generation/exchange_grid_toolset/workdir/mosaic_c96om5b04v20240410.20240423.an105/mosaic_c96om5b04v20240410.20240423.an105.tar"
+  switches:
+    do_timeavgs:                False
+    clean_work:                 False
+    do_refinediag:              False
+    do_atmos_plevel_masking:    False
+    do_preanalysis:             False
+    do_analysis:                False
+    do_analysis_only:           False
+```
+
+It's useful to make sure some of these are set to specific values for developer work:
+
+  `postprocess:switches:clean_work`
+
+  - *what it does:* Whether you clean up the contents of the intermediate directories
```

Comment:
is workdir or tmpdir used more often than intermediate directories?  Maybe swap intermediate for the more often used terminology